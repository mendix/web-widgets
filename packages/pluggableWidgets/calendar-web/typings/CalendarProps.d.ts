/**
 * This file was generated from Calendar.xml
 * WARNING: All changes made to this file will be overwritten
 * @author Mendix Widgets Framework Team
 */
import { CSSProperties } from "react";
import { ActionValue, EditableValue, ListValue, Option, ListActionValue, ListAttributeValue, ListExpressionValue } from "mendix";

export type TitleTypeEnum = "attribute" | "expression";

export type ViewEnum = "standard" | "custom";

export type EditableEnum = "default" | "never";

export type DefaultViewEnum = "day" | "week" | "month" | "work_week" | "agenda";

export type WidthUnitEnum = "pixels" | "percentage";

export type HeightUnitEnum = "percentageOfWidth" | "pixels" | "percentageOfParent" | "percentageOfView";

export type MinHeightUnitEnum = "none" | "pixels" | "percentageOfParent" | "percentageOfView";

export type MaxHeightUnitEnum = "none" | "pixels" | "percentageOfParent" | "percentageOfView";

export type OverflowYEnum = "auto" | "scroll" | "hidden";

export interface CalendarContainerProps {
    name: string;
    class: string;
    style?: CSSProperties;
    tabIndex?: number;
    databaseDataSource?: ListValue;
    titleType: TitleTypeEnum;
    titleAttribute?: ListAttributeValue<string>;
    titleExpression?: ListExpressionValue<string>;
    allDayAttribute?: ListAttributeValue<boolean>;
    startAttribute?: ListAttributeValue<Date>;
    endAttribute?: ListAttributeValue<Date>;
    eventColor?: ListAttributeValue<string>;
    view: ViewEnum;
    editable: EditableEnum;
    enableCreate: boolean;
    showEventDate: boolean;
    defaultView: DefaultViewEnum;
    startDateAttribute?: EditableValue<Date>;
    minHour: number;
    maxHour: number;
    customViewCaption?: any;
    showMonday: boolean;
    showTuesday: boolean;
    showWednesday: boolean;
    showThursday: boolean;
    showFriday: boolean;
    showSunday: boolean;
    showSaturday: boolean;
    eventDataAttribute?: EditableValue<string>;
    onClickEvent?: ListActionValue<{ startDate: Option<Date>; endDate: Option<Date>; allDay: Option<boolean>; title: Option<string> }>;
    onCreateEvent?: ActionValue<{ startDate: Option<Date>; endDate: Option<Date>; allDay: Option<boolean> }>;
    onChange?: ListActionValue<{ oldStart: Option<Date>; oldEnd: Option<Date>; newStart: Option<Date>; newEnd: Option<Date> }>;
    onRangeChange?: ActionValue<{ rangeStart: Option<Date>; rangeEnd: Option<Date>; currentView: Option<string> }>;
    widthUnit: WidthUnitEnum;
    width: number;
    heightUnit: HeightUnitEnum;
    height: number;
    minHeightUnit: MinHeightUnitEnum;
    minHeight: number;
    maxHeightUnit: MaxHeightUnitEnum;
    maxHeight: number;
    overflowY: OverflowYEnum;
    showAllEvents: boolean;
}

export interface CalendarPreviewProps {
    /**
     * @deprecated Deprecated since version 9.18.0. Please use class property instead.
     */
    className: string;
    class: string;
    style: string;
    styleObject?: CSSProperties;
    readOnly: boolean;
    renderMode: "design" | "xray" | "structure";
    translate: (text: string) => string;
    databaseDataSource: {} | { caption: string } | { type: string } | null;
    titleType: TitleTypeEnum;
    titleAttribute: string;
    titleExpression: string;
    allDayAttribute: string;
    startAttribute: string;
    endAttribute: string;
    eventColor: string;
    view: ViewEnum;
    editable: EditableEnum;
    enableCreate: boolean;
    showEventDate: boolean;
    defaultView: DefaultViewEnum;
    startDateAttribute: string;
    minHour: number | null;
    maxHour: number | null;
    customViewCaption: any;
    showMonday: boolean;
    showTuesday: boolean;
    showWednesday: boolean;
    showThursday: boolean;
    showFriday: boolean;
    showSunday: boolean;
    showSaturday: boolean;
    eventDataAttribute: string;
    onClickEvent: {} | null;
    onCreateEvent: {} | null;
    onChange: {} | null;
    onRangeChange: {} | null;
    widthUnit: WidthUnitEnum;
    width: number | null;
    heightUnit: HeightUnitEnum;
    height: number | null;
    minHeightUnit: MinHeightUnitEnum;
    minHeight: number | null;
    maxHeightUnit: MaxHeightUnitEnum;
    maxHeight: number | null;
    overflowY: OverflowYEnum;
    showAllEvents: boolean;
}
