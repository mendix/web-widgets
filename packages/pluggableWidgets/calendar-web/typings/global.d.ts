// TypeScript interfaces for Mendix global objects

export interface MXLocalePatterns {
    date: string;
    datetime: string;
    time: string;
}

export interface MXLocaleDates {
    weekdays: string[];
    shortWeekdays: string[];
    months: string[];
    shortMonths: string[];
    abbreviatedMonths: string[];
    abbreviatedShortMonths: string[];
    dayPeriods: string[];
    eras: string[];
}

export interface MXLocaleNumbers {
    decimalSeparator: string;
    groupingSeparator: string;
    minusSign: string;
}

export interface MXSessionLocale {
    code: string;
    languageTag: string;
    firstDayOfWeek: number;
    dates: MXLocaleDates;
    patterns: MXLocalePatterns;
    numbers: MXLocaleNumbers;
}

export interface MXSessionData {
    locale: MXSessionLocale;
}

export interface MXSession {
    sessionData: MXSessionData;
}

export interface MXGlobalObject {
    mx?: {
        session?: MXSession;
    };
}

declare global {
    interface Window extends MXGlobalObject {}
}
