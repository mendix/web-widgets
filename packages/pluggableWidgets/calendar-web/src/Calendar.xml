<?xml version="1.0" encoding="utf-8" ?>
<widget id="com.mendix.widget.web.calendar.Calendar" pluginWidget="true" offlineCapable="true" supportedPlatform="Web" xmlns="http://www.mendix.com/widget/1.0/" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.mendix.com/widget/1.0/ ../node_modules/mendix/custom_widget.xsd">
    <name>Calendar</name>
    <description>Calendar</description>
    <studioCategory>Display</studioCategory>
    <helpUrl>https://docs.mendix.com/appstore/widgets/calendar</helpUrl>
    <properties>
        <propertyGroup caption="General">
            <!-- OPTIONS SOURCE -->
            <propertyGroup caption="Data source">
                <property key="databaseDataSource" type="datasource" isList="true" required="false">
                    <caption>Events</caption>
                    <description />
                </property>
                <property key="titleType" type="enumeration" defaultValue="attribute">
                    <caption>Title type</caption>
                    <description>Choose between an attribute or an expression for the title</description>
                    <enumerationValues>
                        <enumerationValue key="attribute">Attribute</enumerationValue>
                        <enumerationValue key="expression">Expression</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="titleAttribute" type="attribute" dataSource="databaseDataSource" required="false">
                    <caption>Title attribute</caption>
                    <description>Select an attribute that contains the event title</description>
                    <attributeTypes>
                        <attributeType name="String" />
                    </attributeTypes>
                </property>
                <property key="titleExpression" type="expression" dataSource="databaseDataSource" required="false">
                    <caption>Title expression</caption>
                    <description>Expression that results in the event title</description>
                    <returnType type="String" />
                </property>
                <property key="allDayAttribute" type="attribute" dataSource="databaseDataSource" required="false">
                    <caption>All day attribute</caption>
                    <description />
                    <attributeTypes>
                        <attributeType name="Boolean" />
                    </attributeTypes>
                </property>
                <property key="startAttribute" type="attribute" dataSource="databaseDataSource" required="false">
                    <caption>Start attribute</caption>
                    <description />
                    <attributeTypes>
                        <attributeType name="DateTime" />
                    </attributeTypes>
                </property>
                <property key="endAttribute" type="attribute" dataSource="databaseDataSource" required="false">
                    <caption>End attribute</caption>
                    <description />
                    <attributeTypes>
                        <attributeType name="DateTime" />
                    </attributeTypes>
                </property>
                <property key="eventColor" type="attribute" dataSource="databaseDataSource" required="false">
                    <caption>Color attribute</caption>
                    <description>Attribute containing a valid HTML color eg: red #FF0000 rgb(250,10,20) rgba(10,10,10, 0.5)</description>
                    <attributeTypes>
                        <attributeType name="Enum" />
                        <attributeType name="String" />
                    </attributeTypes>
                </property>
            </propertyGroup>
            <propertyGroup caption="Options">
                <property key="view" type="enumeration" defaultValue="standard">
                    <caption>View</caption>
                    <description>Standard has day, week and month</description>
                    <enumerationValues>
                        <enumerationValue key="standard">Standard</enumerationValue>
                        <enumerationValue key="custom">Custom</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="editable" type="enumeration" defaultValue="default">
                    <caption>Editable</caption>
                    <description />
                    <enumerationValues>
                        <enumerationValue key="default">Default</enumerationValue>
                        <enumerationValue key="never">Never</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="enableCreate" type="boolean" defaultValue="true">
                    <caption>Enable create</caption>
                    <description />
                </property>
                <property key="showEventDate" type="boolean" defaultValue="true">
                    <caption>Show event date range</caption>
                    <description>Show the start and end date of the event</description>
                </property>
                <property key="defaultViewCustom" type="enumeration" defaultValue="month">
                    <caption>Initial selected view</caption>
                    <description>The default view showed when the calendar is loaded</description>
                    <enumerationValues>
                        <enumerationValue key="day">Day</enumerationValue>
                        <enumerationValue key="week">Week</enumerationValue>
                        <enumerationValue key="month">Month</enumerationValue>
                        <enumerationValue key="work_week">Custom</enumerationValue>
                        <enumerationValue key="agenda">Agenda</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="defaultViewStandard" type="enumeration" defaultValue="month">
                    <caption>Initial selected view</caption>
                    <description>The default view showed when the calendar is loaded</description>
                    <enumerationValues>
                        <enumerationValue key="day">Day</enumerationValue>
                        <enumerationValue key="week">Week</enumerationValue>
                        <enumerationValue key="month">Month</enumerationValue>
                    </enumerationValues>
                </property>
                <property key="startDateAttribute" type="attribute" required="false">
                    <caption>Start date attribute</caption>
                    <description>The start date that should be shown in the view</description>
                    <attributeTypes>
                        <attributeType name="DateTime" />
                    </attributeTypes>
                </property>
                <property key="minHour" type="integer" defaultValue="0">
                    <caption>Day start hour</caption>
                    <description>The hour at which the day view starts (0–23)</description>
                </property>
                <property key="maxHour" type="integer" defaultValue="24">
                    <caption>Day end hour</caption>
                    <description>The hour at which the day view ends (1–24)</description>
                </property>
                <!-- Custom week caption -->
                <property key="customViewCaption" type="textTemplate" required="false">
                    <caption>Custom view caption</caption>
                    <description>Label used for the custom work-week button and title. Defaults to "Custom".</description>
                    <translations>
                        <translation lang="en_US">Custom</translation>
                    </translations>
                </property>
            </propertyGroup>
            <propertyGroup caption="Visible days">
                <property key="showMonday" type="boolean" defaultValue="true">
                    <caption>Monday</caption>
                    <description>Show Monday in the custom work-week view</description>
                </property>
                <property key="showTuesday" type="boolean" defaultValue="true">
                    <caption>Tuesday</caption>
                    <description>Show Tuesday in the custom work-week view</description>
                </property>
                <property key="showWednesday" type="boolean" defaultValue="true">
                    <caption>Wednesday</caption>
                    <description>Show Wednesday in the custom work-week view</description>
                </property>
                <property key="showThursday" type="boolean" defaultValue="true">
                    <caption>Thursday</caption>
                    <description>Show Thursday in the custom work-week view</description>
                </property>
                <property key="showFriday" type="boolean" defaultValue="true">
                    <caption>Friday</caption>
                    <description>Show Friday in the custom work-week view</description>
                </property>
                <property key="showSunday" type="boolean" defaultValue="false">
                    <caption>Sunday</caption>
                    <description>Show Sunday in the custom work-week view</description>
                </property>
                <property key="showSaturday" type="boolean" defaultValue="false">
                    <caption>Saturday</caption>
                    <description>Show Saturday in the custom work-week view</description>
                </property>
            </propertyGroup>
        </propertyGroup>
        <propertyGroup caption="Events">
            <property key="eventDataAttribute" type="attribute" required="false">
                <caption>Event data attribute</caption>
                <description>The attribute to store received raw data</description>
                <attributeTypes>
                    <attributeType name="String" />
                </attributeTypes>
            </property>
            <property key="onClickEvent" type="action" required="false" dataSource="databaseDataSource">
                <caption>On click action</caption>
                <description />
                <actionVariables>
                    <actionVariable key="startDate" type="DateTime" caption="Event start date" />
                    <actionVariable key="endDate" type="DateTime" caption="Event end date" />
                    <actionVariable key="allDay" type="Boolean" caption="Event all day" />
                    <actionVariable key="title" type="String" caption="Event title" />
                </actionVariables>
            </property>
            <property key="onCreateEvent" type="action" required="false">
                <caption>On create action</caption>
                <description>The create event is triggered when a time slot is selected, and the 'Enable create' property is set to 'true'</description>
                <actionVariables>
                    <actionVariable key="startDate" type="DateTime" caption="New event start date" />
                    <actionVariable key="endDate" type="DateTime" caption="New event end date" />
                    <actionVariable key="allDay" type="Boolean" caption="All day flag" />
                </actionVariables>
            </property>
            <property key="onChange" type="action" required="false" dataSource="databaseDataSource">
                <caption>On change action</caption>
                <description>The change event is triggered on moving/dragging an item or changing the start or end time of by resizing an item</description>
                <actionVariables>
                    <actionVariable key="oldStart" type="DateTime" caption="Old start date" />
                    <actionVariable key="oldEnd" type="DateTime" caption="Old end date" />
                    <actionVariable key="newStart" type="DateTime" caption="New start date" />
                    <actionVariable key="newEnd" type="DateTime" caption="New end date" />
                </actionVariables>
            </property>
            <property key="onRangeChange" type="action" required="false">
                <caption>On view range change</caption>
                <description>Triggered when the calendar view range (start/end) changes</description>
                <actionVariables>
                    <actionVariable key="rangeStart" type="DateTime" caption="View range start" />
                    <actionVariable key="rangeEnd" type="DateTime" caption="View range end" />
                    <actionVariable key="currentView" type="String" caption="Current view" />
                </actionVariables>
            </property>
        </propertyGroup>
        <propertyGroup caption="Dimensions">
            <property key="widthUnit" type="enumeration" defaultValue="percentage">
                <caption>Width unit</caption>
                <description />
                <enumerationValues>
                    <enumerationValue key="pixels">Pixels</enumerationValue>
                    <enumerationValue key="percentage">Percentage</enumerationValue>
                </enumerationValues>
            </property>
            <property key="width" type="integer" defaultValue="100">
                <caption>Width</caption>
                <description />
            </property>
            <property key="heightUnit" type="enumeration" defaultValue="pixels">
                <caption>Height unit</caption>
                <description />
                <enumerationValues>
                    <enumerationValue key="percentageOfWidth">Auto</enumerationValue>
                    <enumerationValue key="pixels">Pixels</enumerationValue>
                    <enumerationValue key="percentageOfParent">Percentage</enumerationValue>
                    <enumerationValue key="percentageOfView">Viewport</enumerationValue>
                </enumerationValues>
            </property>
            <property key="height" type="integer" defaultValue="580">
                <caption>Height</caption>
                <description />
            </property>
            <property key="minHeightUnit" type="enumeration" defaultValue="pixels">
                <caption>Minimum height unit</caption>
                <description />
                <enumerationValues>
                    <enumerationValue key="none">None</enumerationValue>
                    <enumerationValue key="pixels">Pixels</enumerationValue>
                    <enumerationValue key="percentageOfParent">Percentage</enumerationValue>
                    <enumerationValue key="percentageOfView">Viewport</enumerationValue>
                </enumerationValues>
            </property>
            <property key="minHeight" type="integer" defaultValue="250">
                <caption>Minimum height</caption>
                <description />
            </property>
            <property key="maxHeightUnit" type="enumeration" defaultValue="none">
                <caption>Maximum height unit</caption>
                <description />
                <enumerationValues>
                    <enumerationValue key="none">None</enumerationValue>
                    <enumerationValue key="pixels">Pixels</enumerationValue>
                    <enumerationValue key="percentageOfParent">Percentage</enumerationValue>
                    <enumerationValue key="percentageOfView">Viewport</enumerationValue>
                </enumerationValues>
            </property>
            <property key="maxHeight" type="integer" defaultValue="250">
                <caption>Maximum height</caption>
                <description />
            </property>
            <property key="overflowY" type="enumeration" defaultValue="auto">
                <caption>Vertical overflow</caption>
                <description />
                <enumerationValues>
                    <enumerationValue key="auto">Auto</enumerationValue>
                    <enumerationValue key="scroll">Scroll</enumerationValue>
                    <enumerationValue key="hidden">Hidden</enumerationValue>
                </enumerationValues>
            </property>
            <property key="showAllEvents" type="boolean" defaultValue="true">
                <caption>Show all events</caption>
                <description>Auto-adjust calendar height to display all events without "more" links</description>
            </property>
        </propertyGroup>
    </properties>
</widget>
