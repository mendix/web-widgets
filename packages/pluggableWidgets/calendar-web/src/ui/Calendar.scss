@use "sass:color";

.widget-calendar {
    $cal-form-group-margin-bottom: 15px !default;
    $cal-brand-default: #dddddd;
    $cal-brand-primary: #0595db;
    $cal-brand-danger: #ed1c24 !default;
    $cal-border-color-default: #d7d7d7;
    $cal-color-default-lighter: color.mix($cal-brand-default, white, 20%) !default;
    $cal-color-primary-lighter: color.mix($cal-brand-primary, white, 20%) !default;

    min-height: 350px;
    margin-bottom: var(--form-group-margin-bottom, $cal-form-group-margin-bottom);
    -ms-user-select: none;
    -webkit-user-select: none;
    user-select: none;

    .rbc-calendar {
        width: 100%;
        height: 100%;
    }

    .rbc-event {
        min-height: 20px;
    }

    .rbc-addons-dnd .rbc-addons-dnd-resize-ns-icon {
        cursor: row-resize;
    }

    .rbc-addons-dnd .rbc-addons-dnd-resize-ew-icon {
        cursor: col-resize;
    }

    .calendar-label {
        font-size: var(--font-size-default, 16px);
    }

    .calendar-toolbar {
        display: flex;
        align-items: center;
        justify-content: space-between;
        flex-wrap: wrap;
        padding-bottom: var(--spacing-medium, 10px);
    }

    .rbc-current-time-indicator {
        background-color: var(--brand-danger, $cal-brand-danger);
    }

    .rbc-day-slot .rbc-event {
        border-top-color: var(--border-color-default, $cal-border-color-default);
        border-right-color: var(--border-color-default, $cal-border-color-default);
        border-bottom-color: var(--border-color-default, $cal-border-color-default);
        border-left-color: var(--border-color-default, $cal-border-color-default);
    }

    .rbc-off-range-bg {
        background-color: var(--color-default-lighter, $cal-color-default-lighter);
    }

    .rbc-today {
        background-color: var(--color-primary-lighter, $cal-color-primary-lighter);
    }

    .rbc-event {
        background-color: var(--brand-primary, $cal-brand-primary);
    }
}
