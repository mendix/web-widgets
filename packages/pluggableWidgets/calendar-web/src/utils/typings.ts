import { ObjectItem } from "mendix";
import { CalendarProps as ReactCalendarProps } from "react-big-calendar";
import { withDragAndDropProps } from "react-big-calendar/lib/addons/dragAndDrop";

export interface CalendarEvent {
    title: string;
    start: Date;
    end: Date;
    allDay: boolean;
    color?: string;
    item: ObjectItem;
}

export type EventDropOrResize = {
    event: CalendarEvent;
    start: Date;
    end: Date;
};

export interface DragAndDropCalendarProps<TEvent extends object = Event, TResource extends object = object>
    extends ReactCalendarProps<TEvent, TResource>,
        withDragAndDropProps<TEvent, TResource> {}
