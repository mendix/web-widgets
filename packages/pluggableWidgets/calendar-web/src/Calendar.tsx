import classnames from "classnames";
import { ReactElement, createElement } from "react";
import { CalendarContainerProps } from "../typings/CalendarProps";
import { CalendarPropsBuilder } from "./helpers/CalendarPropsBuilder";
import { DnDCalendar } from "./utils/calendar-utils";
import { constructWrapperStyle } from "./utils/style-utils";
import "./ui/Calendar.scss";

export default function MxCalendar(props: CalendarContainerProps): ReactElement {
    const { class: className } = props;
    const wrapperStyle = constructWrapperStyle(props);
    const calendarProps = new CalendarPropsBuilder(props).build();

    return (
        <div className={classnames("widget-calendar", className)} style={wrapperStyle}>
            <DnDCalendar {...calendarProps} />
        </div>
    );
}
