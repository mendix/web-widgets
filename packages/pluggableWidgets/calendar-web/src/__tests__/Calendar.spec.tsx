import { createElement } from "react";
import { render } from "@testing-library/react";
import { ListValueBuilder } from "@mendix/widget-plugin-test-utils";

import Calendar from "../Calendar";
import { CalendarContainerProps } from "../../typings/CalendarProps";
const customViewProps: CalendarContainerProps = {
    name: "calendar-test",
    class: "calendar-class",
    tabIndex: 0,
    databaseDataSource: new ListValueBuilder().withItems([]).build(),
    titleType: "attribute",
    view: "custom",
    defaultViewStandard: "month",
    defaultViewCustom: "work_week",
    editable: "default",
    enableCreate: true,
    widthUnit: "percentage",
    width: 100,
    heightUnit: "pixels",
    height: 400,
    minHour: 0,
    maxHour: 24,
    minHeightUnit: "pixels",
    minHeight: 400,
    maxHeightUnit: "none",
    maxHeight: 400,
    overflowY: "auto",
    showEventDate: true,
    showSunday: false,
    showMonday: true,
    showTuesday: true,
    showWednesday: true,
    showThursday: true,
    showFriday: true,
    showSaturday: false,
    showAllEvents: true
};

const standardViewProps: CalendarContainerProps = {
    ...customViewProps,
    view: "standard"
};

beforeAll(() => {
    jest.useFakeTimers();
    jest.setSystemTime(new Date("2025-04-28T12:00:00Z"));
});

afterAll(() => {
    jest.useRealTimers();
});

describe("Calendar", () => {
    it("renders correctly with basic props", () => {
        const calendar = render(<Calendar {...customViewProps} />);
        expect(calendar).toMatchSnapshot();
    });

    it("renders with correct class name", () => {
        const { container } = render(<Calendar {...customViewProps} />);
        expect(container.querySelector(".widget-calendar")).toBeTruthy();
        expect(container.querySelector(".calendar-class")).toBeTruthy();
    });

    it("does not render custom view button in standard view", () => {
        const { queryByText } = render(<Calendar {...standardViewProps} />);
        expect(queryByText("Custom")).toBeNull();
    });
});
