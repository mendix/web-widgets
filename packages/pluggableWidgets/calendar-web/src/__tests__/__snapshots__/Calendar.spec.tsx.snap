// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar renders correctly with basic props 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="widget-calendar calendar-class"
        style="width: 100%; height: 400px;"
      >
        <div
          class="rbc-addons-dnd rbc-calendar"
        >
          <div
            class="calendar-toolbar"
          >
            <div
              class="btn-group"
            >
              <button
                class="btn btn-default"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="glyphicon glyphicon-backward"
                />
              </button>
              <button
                class="btn btn-default"
                type="button"
              >
                Today
              </button>
              <button
                class="btn btn-default"
                type="button"
              >
                <span
                  aria-hidden="true"
                  class="glyphicon glyphicon-forward"
                />
              </button>
            </div>
            <div
              class="btn-group"
            >
              <span
                class="calendar-label"
              >
                Apr 28 – May 02
              </span>
            </div>
            <div
              class="btn-group"
            >
              <button
                class="btn btn-default"
                type="button"
              >
                Day
              </button>
              <button
                class="btn btn-default"
                type="button"
              >
                Week
              </button>
              <button
                class="btn btn-default"
                type="button"
              >
                Month
              </button>
              <button
                class="btn btn-default active"
                type="button"
              >
                Custom
              </button>
              <button
                class="btn btn-default"
                type="button"
              >
                Agenda
              </button>
            </div>
          </div>
          <div
            class="rbc-time-view"
          >
            <div
              class="rbc-time-header"
            >
              <div
                class="rbc-label rbc-time-header-gutter"
              />
              <div
                class="rbc-time-header-content"
              >
                <div
                  class="rbc-row rbc-time-header-cell"
                >
                  <div
                    class="rbc-header rbc-today"
                  >
                    <button
                      class="rbc-button-link"
                      type="button"
                    >
                      <span
                        aria-sort="none"
                        role="columnheader"
                      >
                        28 Mon
                      </span>
                    </button>
                  </div>
                  <div
                    class="rbc-header"
                  >
                    <button
                      class="rbc-button-link"
                      type="button"
                    >
                      <span
                        aria-sort="none"
                        role="columnheader"
                      >
                        29 Tue
                      </span>
                    </button>
                  </div>
                  <div
                    class="rbc-header"
                  >
                    <button
                      class="rbc-button-link"
                      type="button"
                    >
                      <span
                        aria-sort="none"
                        role="columnheader"
                      >
                        30 Wed
                      </span>
                    </button>
                  </div>
                  <div
                    class="rbc-header"
                  >
                    <button
                      class="rbc-button-link"
                      type="button"
                    >
                      <span
                        aria-sort="none"
                        role="columnheader"
                      >
                        01 Thu
                      </span>
                    </button>
                  </div>
                  <div
                    class="rbc-header"
                  >
                    <button
                      class="rbc-button-link"
                      type="button"
                    >
                      <span
                        aria-sort="none"
                        role="columnheader"
                      >
                        02 Fri
                      </span>
                    </button>
                  </div>
                </div>
                <div
                  class="rbc-allday-cell"
                  role="rowgroup"
                >
                  <div
                    class="rbc-row-bg"
                  >
                    <div
                      class="rbc-day-bg rbc-today"
                    />
                    <div
                      class="rbc-day-bg"
                    />
                    <div
                      class="rbc-day-bg"
                    />
                    <div
                      class="rbc-day-bg"
                    />
                    <div
                      class="rbc-day-bg"
                    />
                  </div>
                  <div
                    class="rbc-row-content"
                    role="row"
                  >
                    <div
                      class="rbc-addons-dnd-row-body"
                    >
                      <div
                        class="rbc-row"
                      />
                      <div
                        class="rbc-row"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="rbc-time-content"
            >
              <div
                class="rbc-time-gutter rbc-time-column"
              />
              <div
                class="rbc-day-slot rbc-time-column rbc-now rbc-today"
              >
                <div>
                  <div
                    class="rbc-events-container"
                  />
                </div>
              </div>
              <div
                class="rbc-day-slot rbc-time-column"
              >
                <div>
                  <div
                    class="rbc-events-container"
                  />
                </div>
              </div>
              <div
                class="rbc-day-slot rbc-time-column"
              >
                <div>
                  <div
                    class="rbc-events-container"
                  />
                </div>
              </div>
              <div
                class="rbc-day-slot rbc-time-column"
              >
                <div>
                  <div
                    class="rbc-events-container"
                  />
                </div>
              </div>
              <div
                class="rbc-day-slot rbc-time-column"
              >
                <div>
                  <div
                    class="rbc-events-container"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="widget-calendar calendar-class"
      style="width: 100%; height: 400px;"
    >
      <div
        class="rbc-addons-dnd rbc-calendar"
      >
        <div
          class="calendar-toolbar"
        >
          <div
            class="btn-group"
          >
            <button
              class="btn btn-default"
              type="button"
            >
              <span
                aria-hidden="true"
                class="glyphicon glyphicon-backward"
              />
            </button>
            <button
              class="btn btn-default"
              type="button"
            >
              Today
            </button>
            <button
              class="btn btn-default"
              type="button"
            >
              <span
                aria-hidden="true"
                class="glyphicon glyphicon-forward"
              />
            </button>
          </div>
          <div
            class="btn-group"
          >
            <span
              class="calendar-label"
            >
              Apr 28 – May 02
            </span>
          </div>
          <div
            class="btn-group"
          >
            <button
              class="btn btn-default"
              type="button"
            >
              Day
            </button>
            <button
              class="btn btn-default"
              type="button"
            >
              Week
            </button>
            <button
              class="btn btn-default"
              type="button"
            >
              Month
            </button>
            <button
              class="btn btn-default active"
              type="button"
            >
              Custom
            </button>
            <button
              class="btn btn-default"
              type="button"
            >
              Agenda
            </button>
          </div>
        </div>
        <div
          class="rbc-time-view"
        >
          <div
            class="rbc-time-header"
          >
            <div
              class="rbc-label rbc-time-header-gutter"
            />
            <div
              class="rbc-time-header-content"
            >
              <div
                class="rbc-row rbc-time-header-cell"
              >
                <div
                  class="rbc-header rbc-today"
                >
                  <button
                    class="rbc-button-link"
                    type="button"
                  >
                    <span
                      aria-sort="none"
                      role="columnheader"
                    >
                      28 Mon
                    </span>
                  </button>
                </div>
                <div
                  class="rbc-header"
                >
                  <button
                    class="rbc-button-link"
                    type="button"
                  >
                    <span
                      aria-sort="none"
                      role="columnheader"
                    >
                      29 Tue
                    </span>
                  </button>
                </div>
                <div
                  class="rbc-header"
                >
                  <button
                    class="rbc-button-link"
                    type="button"
                  >
                    <span
                      aria-sort="none"
                      role="columnheader"
                    >
                      30 Wed
                    </span>
                  </button>
                </div>
                <div
                  class="rbc-header"
                >
                  <button
                    class="rbc-button-link"
                    type="button"
                  >
                    <span
                      aria-sort="none"
                      role="columnheader"
                    >
                      01 Thu
                    </span>
                  </button>
                </div>
                <div
                  class="rbc-header"
                >
                  <button
                    class="rbc-button-link"
                    type="button"
                  >
                    <span
                      aria-sort="none"
                      role="columnheader"
                    >
                      02 Fri
                    </span>
                  </button>
                </div>
              </div>
              <div
                class="rbc-allday-cell"
                role="rowgroup"
              >
                <div
                  class="rbc-row-bg"
                >
                  <div
                    class="rbc-day-bg rbc-today"
                  />
                  <div
                    class="rbc-day-bg"
                  />
                  <div
                    class="rbc-day-bg"
                  />
                  <div
                    class="rbc-day-bg"
                  />
                  <div
                    class="rbc-day-bg"
                  />
                </div>
                <div
                  class="rbc-row-content"
                  role="row"
                >
                  <div
                    class="rbc-addons-dnd-row-body"
                  >
                    <div
                      class="rbc-row"
                    />
                    <div
                      class="rbc-row"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rbc-time-content"
          >
            <div
              class="rbc-time-gutter rbc-time-column"
            />
            <div
              class="rbc-day-slot rbc-time-column rbc-now rbc-today"
            >
              <div>
                <div
                  class="rbc-events-container"
                />
              </div>
            </div>
            <div
              class="rbc-day-slot rbc-time-column"
            >
              <div>
                <div
                  class="rbc-events-container"
                />
              </div>
            </div>
            <div
              class="rbc-day-slot rbc-time-column"
            >
              <div>
                <div
                  class="rbc-events-container"
                />
              </div>
            </div>
            <div
              class="rbc-day-slot rbc-time-column"
            >
              <div>
                <div
                  class="rbc-events-container"
                />
              </div>
            </div>
            <div
              class="rbc-day-slot rbc-time-column"
            >
              <div>
                <div
                  class="rbc-events-container"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
